CMake/ConfigureCrossCompile.cmake

```diff
@@ -0,0 +1,42 @@
+# This file adds the necessary configurations to cross compile
+# mlpack for embedded systems. You need to set the following variables
+# from the command line: the CMAKE_SYSROOT, TOOLCHAIN_PREFIX and the
+# board type.
+# This file will compile OpenBLAS if it is downloaded and it is not
+# available on you system in order to find the BLAS library.
+
+if (CMAKE_CROSSCOMPILING)
+  include(board/flags-config.cmake)
+  if(NOT CMAKE_SYSROOT AND (NOT TOOLCHAIN_PREFIX))
+    message(FATAL_ERROR "Neither of CMAKE_SYSROOT or TOOLCHAIN_PREFIX is set, please set both of them and try again")
+  elseif(NOT CMAKE_SYSROOT)
+    message(FATAL_ERROR "Can not proceed CMAKE_SYSROOT is not set")
+  elseif(NOT TOOLCHAIN_PREFIX)
+    message(FATAL_ERROR "Cant not proceed TOOLCHAIN_PREFIXN is not set")
+  elseif(NOT OPENBLAS_TARGET)
+    message(FATAL_ERROR "Cant not proceed, board name is not set, please refer to documentation")
```

Comment:
Might be a good idea to include a link to the documentation here, I guess even linking to https://mlpack.org/docs.html would be helpful?

Reply:
That is true, I will add the link :+1: 
But we really need to update documentation later, since there is nothing there related to this one :+1:

Reply:
Actually, I am not sure that not having `OPENBLAS_TARGET` set is really a problem.  That would only be a problem if we need to autodownload and compiled OpenBLAS ourselves.  So perhaps you should move this check there.  Also, I would suggest using an error message that says that `OPENBLAS_TARGET` is not set, and to either set that or `BOARD_NAME` (which will automatically set `OPENBLAS_TARGET`).